lib/matplotlib/_mathtext.py

```diff
@@ -2285,7 +2285,7 @@ def symbol(self, s: str, loc: int,
             if (self._in_subscript_or_superscript or (
                     c in self._binary_operators and (
                     len(s[:loc].split()) == 0 or prev_char == '{' or
-                    prev_char in self._left_delims))):
+                    prev_char in self._left_delims or prev_char == '='))):
```

Comment:
It should not just be =, but all relational operators.

Reply:
good idea, I will change that